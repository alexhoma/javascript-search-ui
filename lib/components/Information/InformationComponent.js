"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
exports.__esModule = true;
var preact_1 = require("preact");
var Template_1 = require("../Template");
/**
 * Result Information Component
 */
var InformationComponent = /** @class */ (function (_super) {
    __extends(InformationComponent, _super);
    /**
     * Constructor
     */
    function InformationComponent(props) {
        var _this = _super.call(this, props) || this;
        _this.state = {
            hits: 0,
            total: 0,
            visible: false
        };
        return _this;
    }
    /**
     * Component will receive props
     *
     * @param props
     */
    InformationComponent.prototype.componentWillReceiveProps = function (props) {
        this.setState(function (prevState) {
            return (props.store.getCurrentResult() == null)
                ? {
                    hits: 0,
                    total: 0,
                    visible: false
                }
                : {
                    hits: props.store.getCurrentResult().getTotalHits(),
                    total: props.store.getCurrentResult().getTotalItems(),
                    visible: true
                };
        });
    };
    InformationComponent.prototype.render = function () {
        var props = this.props;
        var containerClassName = props.classNames.container;
        var containerTemplate = props.template.container;
        var formatData = props.formatData;
        if (!this.state.visible) {
            return;
        }
        var currentQuery = this.props.store.getCurrentQuery();
        var size = currentQuery.getSize();
        var page = currentQuery.getPage();
        var from = (page - 1) * size;
        var to = from + size;
        var totalHits = this.state.hits.toLocaleString();
        if (totalHits === "10,000") {
            totalHits = '+10,000';
        }
        /**
         * Data accessible to the template
         */
        var reducedTemplateData = {
            total_hits: totalHits,
            total_items: this.state.total.toLocaleString(),
            page: page,
            size: size,
            from: from + 1,
            to: to
        };
        var formattedTemplateData = formatData(reducedTemplateData);
        return (preact_1.h(Template_1["default"], { template: containerTemplate, data: formattedTemplateData, className: "as-information " + containerClassName, dictionary: this.props.dictionary }));
    };
    return InformationComponent;
}(preact_1.Component));
InformationComponent.defaultProps = {
    classNames: {
        container: ''
    },
    template: {
        container: 'Found {{total_hits}}/{{total_items}}'
    },
    formatData: function (data) { return data; }
};
exports["default"] = InformationComponent;
