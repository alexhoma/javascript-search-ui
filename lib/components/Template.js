"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
exports.__esModule = true;
var preact_1 = require("preact");
var Hogan = require("hogan.js");
/**
 * Template
 */
var Template = /** @class */ (function (_super) {
    __extends(Template, _super);
    function Template() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        /**
         * Render template
         *
         * @param template
         * @param result
         *
         * @return {any}
         */
        _this.renderTemplate = function (template, result) {
            /**
             * Compile template using hogan.js
             */
            var compiledTemplate = Hogan.compile(template);
            var output = compiledTemplate.render(result);
            return {
                __html: output
            };
        };
        return _this;
    }
    /**
     * Render
     *
     * @return {any}
     */
    Template.prototype.render = function () {
        var props = this.props;
        var template = props.template;
        var data = props.data;
        var className = props.className;
        return (template)
            ? preact_1.h("div", { className: className, dangerouslySetInnerHTML: this.renderTemplate(template, data) })
            : null;
    };
    return Template;
}(preact_1.Component));
exports["default"] = Template;
