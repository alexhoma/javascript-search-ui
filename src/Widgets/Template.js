/** @jsx h */
import { h, Component } from 'preact';
import Hogan from 'hogan.js';

class Template extends Component {
    renderTemplate = (template, data) => {
        let compiledTemplate = Hogan.compile(template);
        let output = compiledTemplate.render(data);


        return {
            __html: output
        };
    };

    render() {
        const {
            template,
            data,
            className
        } = this.props;

        return (template)
            ? <div
                className={className}
                dangerouslySetInnerHTML={this.renderTemplate(template, data)}
            />
            : null
        ;
    }
}

export default Template;