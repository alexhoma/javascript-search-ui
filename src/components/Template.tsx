import { h, Component } from 'preact';
import {TemplateProps} from "./TemplateProps";
import * as Mustache from 'mustache';

/**
 * Template
 */
class Template extends Component<TemplateProps> {

    /**
     * Render template
     *
     * @param template
     * @param result
     * @param dictionary
     *
     * @return {any}
     */
    renderTemplate = (template, result, dictionary) => {

        let trans = function() {
            return function(text, render) {
                return dictionary[text] ?? text;
            }
        }

        let output = Mustache.render(template, {...result, ...{
            'trans': trans
        }});

        return {
            __html: output
        };
    };

    /**
     * Render
     *
     * @return {any}
     */
    render() {

        const props = this.props;
        const template = props.template;
        const data = props.data;
        const className = props.className;
        const dictionary = props.dictionary ?? {};

        return (template)
            ? <div
                className={className}
                dangerouslySetInnerHTML={this.renderTemplate(template, data, dictionary)}
            />
            : null
        ;
    }
}

export default Template;
