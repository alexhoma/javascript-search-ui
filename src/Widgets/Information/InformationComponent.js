/**
 * @jsx h
 */
import { h, Component } from 'preact';
import Template from "../Template";

/**
 * Result Information Component
 */
class InformationComponent extends Component {
    render() {
        const {
            classNames: {
                container: containerClassName
            },
            template: {
                container: containerTemplate
            },
            formatData,
            data
        } = this.props;

        /**
         * Data accessible to the template
         */
        let reducedTemplateData = {
            total_hits: parseInt(data.total_hits).toLocaleString('de-DE'),
            total_items: parseInt(data.total_items).toLocaleString('de-DE')
        };

        let formattedTemplateData = formatData(reducedTemplateData);

        return (
            <Template
                template={containerTemplate}
                data={formattedTemplateData}
                className={`as-information ${containerClassName}`}
            />
        )
    }
}

InformationComponent.defaultProps = {
    classNames: {
        container: ''
    },
    template: {
        container: 'Found {{total_hits}} of {{total_items}} items'
    },
    formatData: data => data
};

export default InformationComponent;