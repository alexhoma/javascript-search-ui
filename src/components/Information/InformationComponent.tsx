import { h, Component } from 'preact';
import Template from "../Template";
import {InformationProps} from "./InformationProps";

/**
 * Result Information Component
 */
class InformationComponent extends Component<InformationProps> {
    render() {

        const props = this.props;
        const containerClassName = props.classNames.container;
        const containerTemplate = props.template.container;
        const formatData = props.formatData;
        const currentResult = props.currentResult;

        /**
         * Data accessible to the template
         */
        let reducedTemplateData = {
            total_hits: currentResult.getTotalHits().toLocaleString(),
            total_items: currentResult.getTotalItems().toLocaleString()
        };

        let formattedTemplateData = formatData(reducedTemplateData);

        return (
            <Template
                template={containerTemplate}
                data={formattedTemplateData}
                className={`as-information ${containerClassName}`}
            />
        )
    }
}

InformationComponent.defaultProps = {
    classNames: {
        container: ''
    },
    template: {
        container: 'Found {{total_hits}}/{{total_items}}'
    },
    formatData: data => data
};

export default InformationComponent;