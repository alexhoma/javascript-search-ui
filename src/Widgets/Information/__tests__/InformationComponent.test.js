/** @jsx h */
import {h} from 'preact';
import {deep, shallow} from 'preact-render-spy';
import InformationComponent from './../InformationComponent';
import Template from '../../Template';

describe('<InformationComponent />',() => {
    const defaultProps = {
        target: '#some-tagret',
        template: {
            container: `
                <span class="total-hits">{{total_hits}}</span>
                <span class="total-items">{{total_items}}</span>
            `
        }
    };
    const data = {
        total_hits: 12,
        total_items: 32
    }

    it('should render with default props', () => {
        const tree = deep(
            <InformationComponent
                {...defaultProps}
                data={data}
            />
        );

        expect(tree).toMatchSnapshot();
    });
    it('should render with custom props', () => {
        /**
         * Full custom props available object
         */
        const customClassNames = {
            classNames: {
                container: 'custom-container-class',
            }
        };

        const tree = deep(
            <InformationComponent
                {...defaultProps}
                {...customClassNames}
                data={data}
            />
        );

        expect(tree).toMatchSnapshot();
    });
    it('should update content when passing new data', () => {
        const tree = deep(
            <InformationComponent
                {...defaultProps}
                data={data}
            />
        );

        const updatedDataProps = {
            total_hits: 34,
            total_items: 67
        }
        tree.render(
            <InformationComponent
                {...defaultProps}
                data={updatedDataProps}
            />
        )

        expect(tree).toMatchSnapshot();
    });
});