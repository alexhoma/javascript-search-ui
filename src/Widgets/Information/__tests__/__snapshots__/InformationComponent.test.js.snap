// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InformationComponent /> should render with custom props 1`] = `
preact-render-spy (1 nodes)
-------
<div class="asui-information custom-container-class">
  
                  <span class="total-hits">12</span>
                  <span class="total-items">32</span>
              
</div>

`;

exports[`<InformationComponent /> should render with default props 1`] = `
preact-render-spy (1 nodes)
-------
<div class="asui-information ">
  
                  <span class="total-hits">12</span>
                  <span class="total-items">32</span>
              
</div>

`;

exports[`<InformationComponent /> should update content when passing new data 1`] = `
preact-render-spy (1 nodes)
-------
<div class="asui-information ">
  
                  <span class="total-hits">34</span>
                  <span class="total-items">67</span>
              
</div>

`;
